akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -63,6 +72,58 @@ private[http2] class Http2ServerDemux(http2Settings: Http2CommonSettings, initia
   def wrapTrailingHeaders(headers: ParsedHeadersFrame): Option[ByteString] = None
 }
 
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[http2] object ConfigurablePing {
+  case object Tick
+  val Ping = PingFrame(false, ByteString("abcdefgh"))
```

Comment:
We can use hardcoded data because we require ping-timeout to be smaller than ping-interval. If/when we remove that restriction, using hardcoded `data` will break ping ACK's.
It'd be good to add a comment mentioning it.

Reply:
Hmm, I realized there are more details on this implementation that rely on having only one ping in-flight. I guess a generic comment about it (not just focused on `data`) could be useful.